packages/db/src/lib/granule.ts

```diff
@@ -354,3 +360,61 @@ export const getGranulesByGranuleId = async (
     .where({ granule_id: granuleId });
   return records;
 };
+
+/**
+ * Change a granules' PG record and its files' PG record based on collection move
+ *
+ * @param {Knex | Knex.Transaction} knexOrTransaction - DB client or transaction
+ * @param {Array<ApiGranuleRecord>} [granules] - updated ApiGranule records
+ * @returns {Promise<void>}
+ */
+export const updateGranulesAndFiles = async (
+  knexOrTransaction: Knex | Knex.Transaction,
+  granules: Array<ApiGranuleRecord>
+):Promise<void> => {
+  const {
+    granules: granulesTable,
+    files: filesTable,
+  } = TableNames;
+
+  try {
+    await createRejectableTransaction(knexOrTransaction, async (trx) => {
```

Comment:
Is it appropriate/intended for the batch update to be all or nothing?  

Nit: This should probably be documented in the header